src/Shared/AppModels/AppDebug.cs

```diff
@@ -0,0 +1,167 @@
+    /// </summary>
+    public partial class AppDebug : ObservableObject
+    {
+        private SemaphoreSlim _semaphoreSlim = new SemaphoreSlim(1, 1);
+
+        [ObservableProperty]
+        private string? _windowSizeStr;
+
+        [ObservableProperty]
+        private string? _memoryUsage;
+
+        /// <summary>
+        /// Holds the list of all logs.
+        /// </summary>
+        public ObservableCollection<string>? AppLogs { get; } = new ObservableCollection<string>();
+
+        /// <summary>
+        /// Command to delete user settings and data.
+        /// </summary>
+        public ICommand PerformUserDataDeletionCommand { get; private set; }
+
+        /// <summary>
+        /// Creates a new instance of <see cref="AppDebug"/>.
+        /// </summary>
+        public AppDebug()
+        {
+            Logger.MessageReceived += Logger_MessageReceived;
+            Window.Current.SizeChanged += Current_SizeChanged;
+            MemoryManager.AppMemoryUsageIncreased += MemoryManager_MemoryChanged;
+            MemoryManager.AppMemoryUsageDecreased += MemoryManager_MemoryChanged;
+            PerformUserDataDeletionCommand = new RelayCommand(PerformUserDataDeletion);
+            UpdateMemoryUsage();
+            SetWindowSize();
+        }
+
+        private void MemoryManager_MemoryChanged(object sender, object e)
+        {
+            UpdateMemoryUsage();
+        }
+
+        private void UpdateMemoryUsage()
+        {
+            var memoryUsed = MemoryManager.AppMemoryUsage;
+            var totalMemory = MemoryManager.AppMemoryUsageLimit;
+            MemoryUsage = $"App Memory used: {SizeSuffix((long)memoryUsed)}, Expected app memory usage limit: {SizeSuffix((long)totalMemory)}.";
+        }
+
+        private void Current_SizeChanged(object sender, Windows.UI.Core.WindowSizeChangedEventArgs e) => SetWindowSize();
+
+        private void SetWindowSize()
+        {
+            WindowSizeStr = $"Current size is: Height: {Window.Current.Bounds.Height}, Width: {Window.Current.Bounds.Width}";
+        }
+
+        private async void Logger_MessageReceived(object sender, LoggerMessageEventArgs e)
+        {
+            await _semaphoreSlim.WaitAsync();
+
+            var formatedMessage = LogFormatter.GetFormattedLogMessage(e);
+
+            if (AppLogs == null)
+                return;
+
+            await CoreApplication.MainView.CoreWindow.Dispatcher.RunAsync(CoreDispatcherPriority.Normal,
+            () =>
+            {
+                AppLogs.Add(formatedMessage);
+            });
+
+            _semaphoreSlim.Release();
+        }
+
+        [RelayCommand]
+        private async void OpenLogFolder()
+        {
+            var logsLocation = LogFormatter.LogFolderPath;
+            var folder = await StorageFolder.GetFolderFromPathAsync(logsLocation);
+
+            await Launcher.LaunchFolderAsync(folder);
+        }
+
+        [RelayCommand]
+        private async void DeleteUserData()
```

Comment:
Don't use `async void` unless you're writing an event handler (this is not an event handler, this is a normal method).

Use `async Task` instead, and `[RelayCommand]` will generate an `AsyncRelayCommand` you can bind to.